spec/requests/api/graphql/queries/offers_spec.rb

```diff
@@ -0,0 +1,200 @@
+        "gravityPartnerId: \"#{
+          partner.gravity_partner_id
+        }\", states: [\"sent\"]"
+      end
+
+      it 'returns only the sent offers' do
+        post '/api/graphql', params: { query: query }, headers: headers
+
+        expect(response.status).to eq 200
+        body = JSON.parse(response.body)
+
+        edges = body.dig('data', 'offers', 'edges')
+        expect(edges.count).to eq 1
+      end
+    end
+
+    describe 'sorting' do
+      let!(:offer) do
+        ps = Fabricate :partner_submission, partner: partner
+        Fabricate :offer, partner_submission: ps, id: 7, created_at: 1.day.ago
+      end
+      let!(:middle_offer) do
+        ps = Fabricate :partner_submission, partner: partner
+        Fabricate :offer, partner_submission: ps, id: 8, created_at: 3.days.ago
+      end
+      let!(:last_offer) do
+        ps = Fabricate :partner_submission, partner: partner
+        Fabricate :offer, partner_submission: ps, id: 9, created_at: 2.days.ago
+      end
+
+      let(:query) do
+        <<-GRAPHQL
+        query {
+          offers(#{query_inputs}) {
+            edges {
+              node {
+                id
+                createdAt
+              }
+            }
+          }
+        }
+        GRAPHQL
+      end
+
+      context 'without a sort column' do
+        let(:query_inputs) do
+          "gravityPartnerId: \"#{partner.gravity_partner_id}\", sort: null"
+        end
+
+        it 'returns the offers sorted ascending by the id column' do
+          post '/api/graphql', params: { query: query }, headers: headers
+
+          expect(response.status).to eq 200
+          body = JSON.parse(response.body)
+
+          edges = body.dig('data', 'offers', 'edges')
+          ids = edges.map { |edge| edge.dig('node', 'id') }
+          expect(ids).to eq %w[9 8 7]
+        end
+      end
+
+      context 'with a valid sort column' do
+        let(:query_inputs) do
+          "gravityPartnerId: \"#{
+            partner.gravity_partner_id
+          }\", sort: CREATED_AT_ASC"
+        end
+
+        it 'returns the offers sorted ascending by that column' do
+          post '/api/graphql', params: { query: query }, headers: headers
+
+          expect(response.status).to eq 200
+          body = JSON.parse(response.body)
+
+          edges = body.dig('data', 'offers', 'edges')
+          ids = edges.map { |edge| edge.dig('node', 'id') }
+          expect(ids).to eq %w[8 9 7]
+        end
+      end
+
+      context 'with a descending direction prefix' do
+        let(:query_inputs) do
+          "gravityPartnerId: \"#{
+            partner.gravity_partner_id
+          }\", sort: CREATED_AT_DESC"
+        end
+
+        it 'returns the offers sorted descending by that column' do
+          post '/api/graphql', params: { query: query }, headers: headers
+
+          expect(response.status).to eq 200
+          body = JSON.parse(response.body)
+
+          edges = body.dig('data', 'offers', 'edges')
```

Comment:
I like this `dig` -- part of ruby lang?

Reply:
Me toooo - docs here:

https://ruby-doc.org/core-2.7.0/Hash.html#method-i-dig